### DISCUSSION TITLE
initrd flashing to USB Storage failed on Orin AGX

### DISCUSSION THREAD

**Discussion by jacklanAetina** [OE4T/meta-tegra#1397] [Q&A]
Hi all

We want flash Yocto OS to USB follow below website , but when run ./initrd-flash to step 2 will display "ERR: RCM boot failed at ......"

Check log error return value 19 , I am not sure what error happen , any one can help?

meta-yocto on kirkstone branch, Error log on attachment file.

https://github.com/OE4T/meta-tegra/wiki/initrd-flashing-support

[   0.0120 ] BR_CID: 0x80012344705DD842240000000E000040
[   0.0180 ] Sending bct_br
[   0.0230 ] Sending mb1
[   0.0238 ] Sending psc_bl1
[   0.0376 ] Sending bct_mb1
[   0.0443 ] Generating blob for T23x
[   0.0473 ] tegrahost_v2 --chip 0x23 0 --generateblob blob.xml blob.bin
[   0.0487 ] Stat for blob_kernel_tegra234-p3701-0000-p3737-0000.dtb failed
Error: Return value 19
Command tegrahost_v2 --chip 0x23 0 --generateblob blob.xml blob.bin
ERR: RCM boot failed at 2023-10-12T08:54:20+08:00

[initrd_flash_usb_error.log](https://github.com/OE4T/meta-tegra/files/12876009/initrd_flash_usb_error.log)

**Comment by madisongh** [comment]
The log shows
```
Board ID(3701) version(500) sku(0004) revision(G.0)
```
The 64GB AGX Orin is SKU 0000, so I'm guessing that SKU 0004 is for the 32GB module?  If so, I don't think support for that module has been tested.  We might need a new machine configuration for it, like we do for the different Orin NX/Nano modules, if the differences are significant.  Otherwise, the flash helper script might need updating in places to rewrite `3701-0000` with `3701-0004` for that SKU.

**Reply by jacklanAetina** [reply]
Yes , we use Orin AGX 32GB Module

**Comment by madisongh** [comment]
It's been a while, but I've recently added new machine configurations for handling the different AGX Orin variants, similar to what we have for handling the different Orin NX SKUs.  For an AGX Orin 32GB SKU installed in the AGX Orin dev kit carrier, use `MACHINE="p3737-0000-p3701-0004"`.  I don't have that specific SKU to test against, but the configuration should be choosing the correct PMIC config file and kernel device tree to use for that particular variant.
